src/usr/local/buildpack/utils/environment.sh

```diff
@@ -72,24 +70,27 @@ function get_tool_version_env () {
 
 function setup_env_files () {
   # env helper, loads tool specific env
+  local install_dir
+  install_dir=$(get_install_dir)
+
   cat >> "$ENV_FILE" <<- EOM
-export BUILDPACK=1 USER_NAME="${USER_NAME}" USER_ID="${USER_ID}" USER_HOME="/home/${USER_NAME}"
+export BUILDPACK=1 USER_NAME="${USER_NAME}" USER_ID="${USER_ID}" USER_HOME="${USER_HOME}"
 
 # openshift override unknown user home
-if [ "\${EUID}" != 0 ]; then
+if [[ "\${EUID}" -ne 0 ]]; then
   export HOME="\${USER_HOME}"
 fi
 
-if [ -d /usr/local/env.d ]; then
-  for i in /usr/local/env.d/*.sh; do
+if [ -d "${install_dir}/env.d" ]; then
+  for i in "${install_dir}/env.d/"*.sh; do
```

Comment:
I think this should be hard coded. this can't change after buildpack install.
```suggestion
if [ -d "${ROOT_DIR}/env.d" ]; then
  for i in "${ROOT_DIR}/env.d/"*.sh; do
```

Or is this replaced before writing to file?

Reply:
This is replaced when writing to the file, otherwise I would also need to quote it.
Reason for extracting is so that it can be tested.

Reply:
```
export BUILDPACK=1 USER_NAME="user" USER_ID="1000" USER_HOME="/home/user"

# openshift override unknown user home
if [ "${EUID}" != 0 ]; then
  export HOME="${USER_HOME}"
fi

if [ -d "/usr/local/env.d" ]; then
  for i in "/usr/local/env.d/"*.sh; do
    if [ -r $i ]; then
      . $i
    fi
  done
  unset i
fi
if [ -d "/home/user"/env.d ]; then
  for i in "/home/user"/env.d/*.sh; do
    if [ -r $i ]; then
      . $i
    fi
  done
  unset i
fi
export PATH="/home/user/bin:${PATH}"
export DEBIAN_FRONTEND=${DEBIAN_FRONTEND-noninteractive}
export LC_ALL=${LC_ALL-C.UTF-8}
export LANG=${LANG-C.UTF-8}
```
This is what it looks like in the docker image

Reply:
`"/home/user"/env.d` looks ugly, should be changed to `"/home/user/env.d"`